tests/unit/multiclass/KNN_unittest.cc

```diff
@@ -91,7 +91,7 @@ TEST(KNN, kdtree_solver)
 
 	int32_t k=4;
 	CEuclideanDistance* distance = new CEuclideanDistance();	
-	CKNN* knn=new CKNN (k, distance, labels, KNN_BRUTE);
+	CKNN* knn=new CKNN (k, distance, labels, KNN_KDTREE);
```

Comment:
hmmm, but the purpose of this pr is about add kdtree test. The rest of code is about fix the error I got after add that test like memory leaking issue and out boundary error. So I think keep this change in this pr would be better.